### PR TITLE
Enable the opt-in EncodingWarning

### PR DESCRIPTION
**Pull request by AA-Turner**
Whilst testing for #10352, building Sphinx failed for me on windows as the new `linkify_issues_in_changelog` uses the default encoding, which isn't UTF-8 on windows. This adds the new [opt-in EncodingWarning
](https://docs.python.org/3/library/io.html#io-encoding-warning) to check for such cases in the future.

A

### Feature or Bugfix
- Refactoring

### Relates
- https://docs.python.org/3/library/io.html#io-encoding-warning

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -66,11 +66,11 @@ doclinter:
 
 .PHONY: test
 test:
-	@$(PYTHON) -X dev -m pytest -v $(TEST)
+	@$(PYTHON) -X dev -X warn_default_encoding -m pytest -v $(TEST)
```
Comment (tk0miya) [inline Makefile:69]: Is it okay to pass this option to the old pythons?

**Review by tk0miya — APPROVED** [review]
LGTM with nits.

File: Makefile (hunk 1 above)
Comment (AA-Turner) [reply Makefile:69]: Yes, see https://docs.python.org/3/using/cmdline.html#cmdoption-X

> It also allows passing arbitrary values and retrieving them through the [sys._xoptions](https://docs.python.org/3/library/sys.html#sys._xoptions) dictionary.

So for Python 3.9 and earlier the value is just ignored.

A

Comment (tk0miya) [reply Makefile:69]: Thank you for your explanation. Okay, let's merge :-)

**Comment by tk0miya** [comment]
Note: fixed a conflict.

File: Makefile (hunk 1 above)
Comment (AA-Turner) [reply Makefile:69]: Thanks!

**Comment by AA-Turner** [comment]
CI passed here too!

**Comment by tk0miya** [comment]
Merged!

**Pull request merged**